src/raqm.c

```diff
@@ -1077,6 +1083,159 @@ raqm_set_freetype_load_flags_range (raqm_t *rq,
   return _raqm_set_freetype_load_flags (rq, flags, start, end);
 }
 
+/* CSS  Word seperators, word spacing is only applied on these.*/
+static size_t word_separators_len = 7;
+static const uint32_t word_separators[] =
+{
+  0x0020,  /// Space
+  0x00A0,  /// No break space
+  0x1361,  /// Ethiopic word space
+  0x10100, /// Aegean wordspace
+  0x10101, /// Aegean wordspace
+  0x1039F, /// Ugaric word divider
+  0x1091F  /// Phoenician word separator
+};
+
+static bool
+_raqm_set_spacing (raqm_t *rq,
+                   int    spacing,
+                   bool   percentage,
+                   bool   word_spacing,
+                   size_t start,
+                   size_t end)
+{
+  if (!rq)
+    return false;
+
+  if (!rq->text_len)
+    return true;
+
+  if (start >= rq->text_len || end > rq->text_len)
+    return false;
+
+  if (!rq->text_info)
+    return false;
+
+  for (size_t i = start; i < end; i++)
+    if (word_spacing)
+    {
+      for (size_t j = 0; j < word_separators_len; j++)
+      {
+        if (rq->text[i] == word_separators[j])
+        {
+          rq->text_info[i].spacing_after = spacing;
+          rq->text_info[i].spacing_is_percentage = percentage;
+        }
+      }
```

Comment:
```c
uint32_t *sep = word_separators;
...
  while (*(++sep)) {
    if (rq->text[i] == *sep)
    {
      ...
    }
  }

Reply:
Actually, since there are only 8 of them, I wounder if a simple if statement would be better:
```C
  if (rq->text[i] == 0x0020 || /* Space */
      rq->text[i] == 0x00A0 || /* No break space */
      ...)
  {
     ...
  }

Reply:
I went with the if statement, as it is probably more simple. I initially used the current method because the same is done for the list of brackets for script detection.